Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -221,149 +262,258 @@ public List<InteractableEvent> InteractableEvents
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.Focus) > 0; }
+            set
+            {
+                if (HasFocus != value)
+                {
+                    if (!value && HasPress)
+                    {
+                        rollOffTimer = 0;
+                    }
+                    else
+                    {
+                        rollOffTimer = rollOffTime;
+                    }
+
+                    SetState(InteractableStates.InteractableStateEnum.Focus, value);
+                }
+            }
+        }
 
         /// <summary>
-        /// Is disabled
+        /// Currently being pressed
         /// </summary>
-        public bool IsDisabled { get; private set; }
+        public virtual bool HasPress
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.Pressed) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.Pressed, value); }
+        }
 
-        // advanced button states from InteractableStates.InteractableStateEnum
         /// <summary>
-        /// Has focus, finger up - custom: not set by Interactable
+        /// Targeted means the item has focus and finger is up
+        /// Currently not controlled by Interactable directly
         /// </summary>
-        public bool IsTargeted { get; private set; }
+        public virtual bool IsTargeted
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.Targeted) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.Targeted, value); }
+        }
 
         /// <summary>
-        /// No focus, finger is up - custom: not set by Interactable
+        /// State that corresponds to no focus,and finger is up.
+        /// Currently not controlled by Interactable directly
         /// </summary>
-        public bool IsInteractive { get; private set; }
+        public virtual bool IsInteractive
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.Interactive) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.Interactive, value); }
+        }
 
         /// <summary>
-        /// Has focus, finger down - custom: not set by Interactable
+        /// State that corresponds to has focus,and finger down.
+        /// Currently not controlled by Interactable directly
         /// </summary>
-        public bool HasObservationTargeted { get; private set; }
+        public virtual bool HasObservationTargeted
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.ObservationTargeted) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.ObservationTargeted, value); }
+        }
 
         /// <summary>
-        /// No focus, finger down - custom: not set by Interactable
+        /// State that corresponds to no focus,and finger is down.
+        /// Currently not controlled by Interactable directly
```

Comment:
is there a use case?

Reply:
No these auxiliary states I don't understand very well. These aren't priority for most users but I don't want to eliminate their access